src/Microsoft.AspNetCore.Mvc.TagHelpers/ValidationMessageTagHelper.cs

```diff
@@ -71,11 +71,13 @@ public ValidationMessageTagHelper(IHtmlGenerator generator)
                     };
                 }
 
+                var childContent = await output.GetChildContentAsync();
+
                 var tagBuilder = Generator.GenerateValidationMessage(
                     ViewContext,
                     For.ModelExplorer,
                     For.Name,
-                    message: null,
+                    message: childContent.IsEmptyOrWhiteSpace ? null : childContent.GetContent(),
```

Comment:
This condition also needs to check `output.IsContentModified` and to pass `message: null` in that case.